payments_service/braintree/views.py

```diff
@@ -129,22 +135,114 @@ def render(self, data, media_type=None, renderer_context=None):
+        except HttpClientError, exc:
+            log.info('webhook GET: ignoring bad request: '
+                     '{e.__class__.__name__}: {e}'.format(e=exc))
+            result = 'verification failed'
+            status = 400
+        return Response(result, status=status,
+                        content_type='text/plain; charset=utf-8')
+
+    def post(self, request):
+        try:
+            webhook_result = self.api.braintree.webhook.post({
+                'bt_payload': request.DATA.get('bt_payload'),
+                'bt_signature': request.DATA.get('bt_signature'),
+            })
+            self.notify_user_of_webhook(webhook_result)
+            result = ''
+            status = 200
+        except HttpClientError, exc:
+            log.info('webhook POST: ignoring bad request: '
+                     '{e.__class__.__name__}: {e}'.format(e=exc))
+            result = 'verification failed'
+            status = 400
+
+        # Even though we might be returning an empty response,
+        # Braintree expects this to be a 200.
+        return Response('', status=status)
+
+    def notify_user_of_webhook(self, result):
+        log.debug('about to handle subscription: {s}'
+                  .format(s=result))
+
+        kind = result['braintree']['kind']
+        if kind !=  'subscription_charged_successfully':
+            log.warning('not notifying user of webhook kind: {k}'
+                        .format(k=kind))
+            return
+
+        # TODO: get the real product.
+        # https://github.com/mozilla/solitude/issues/481
+        product = products['mozilla-concrete-brick']
+
+        # TODO: use a real lookup in the config
+        # https://github.com/mozilla/payments-config/issues/8
+        sellers_by_product = {}
+        for seller_id, seller in sellers.items():
+            for p in seller.products:
+                sellers_by_product[p.id] = seller
+
+        bt_trans = result['mozilla']['transaction']['braintree']
+
+        # TODO: send an HTML email.
+        # https://github.com/mozilla/payments-service/issues/59
+        # TODO: get the actual seller name, not the ID.
+        # https://github.com/mozilla/payments-config/issues/7
+        # TODO: I think we need to get the actual price of the transaction here
+        # (localized).
+        # TODO: This assumes monthly subscriptions but we should figure that
+        # out automatically.
+        # TODO: link to terms and conditions for the payment.
+        body = textwrap.dedent("""
+
+        {seller.id}
+        {seller.url}
+
+        Date:                           {today_date}
+        Receipt #                       {transaction[uuid]}
+        Paid with                       {cc_type} ending in {cc_truncated_id}
+        {product.description}           {product.price[en]}
+        {bill_start} - {bill_end}       per month
+
+        TOTAL: {product.price[en]}
+        Next payment: {next_pay_date}
+
+        """).format(
```

Comment:
This email string is just temporary. We can set up templating in the HTML patch -- this will require a pile of other changes since there is no HTML rendering in the service yet.